core-spark-adapter-2.4/src/test/scala/za/co/absa/spline/coresparkadapterapi/WriteCommandParserImplSpec.scala

```diff
@@ -34,9 +37,36 @@ class WriteCommandParserImplSpec extends FunSpec with BeforeAndAfterEach with Ma
       when(command.fileFormat).thenReturn(new TextFileFormat())
       val query = mock[LogicalPlan]
       when(command.query).thenReturn(query)
-      WriteCommandParser.instance.matches(command) shouldBe true
-      val writeCommand = WriteCommandParser.instance.asWriteCommand(command)
+      val factory = new WriteCommandParserFactoryImpl()
+      val instance = factory.getWriteParser()
+      instance.matches(command) shouldBe true
+      val writeCommand = instance.asWriteCommand(command)
       writeCommand shouldBe WriteCommand("path1", SaveMode.Append, "Text", query)
     }
   }
+
+  describe("SaveAsTableCommandParserImpl") {
+    it("asWriteCommand") {
+      val tableIdentifier = mock[TableIdentifier]
+      when(tableIdentifier.identifier).thenReturn("tableIdentifier")
+
+      val mockTable = mock[CatalogTable]
+      when(mockTable.identifier).thenReturn(tableIdentifier)
+
+      val storage = mock[CatalogStorageFormat]
+      when(mockTable.storage).thenReturn(storage)
+
+      val command = mock[CreateDataSourceTableAsSelectCommand]
+      when(command.table).thenReturn(mockTable)
+      when(command.mode).thenReturn(SaveMode.Append)
+
+      val query = mock[LogicalPlan]
+      when(command.query).thenReturn(query)
+      val factory = new WriteCommandParserFactoryImpl()
+      val instance = factory.getSaveAsTableParser()
+      instance.matches(command) shouldBe true
+      val writeCommand = instance.asWriteCommand(command)
+      writeCommand shouldBe SaveAsTableCommand("table:/tableIdentifier", SaveMode.Append, "table", query)
```

Comment:
Shoudn't the first param be URL? Then I would expect two slashes after "table" scheme:
```URI = scheme:[//authority]path[?query][#fragment]```